Import-VM-BlogPost

DirectedSoul1 opened the pull request:
Signed-off-by: shegde <shegde@redhat.com>

iranzo commented on _posts/2019-05-15-How-To-Guide-Import-VM-Kubevirt.markdown:
```diff
@@ -0,0 +1,448 @@
+
+**Connect to VirtualMachineInstance console**
+
+Use virtctl to connect to the newly create VirtualMachinInstance.
```
'Machine'

DirectedSoul1 replied:
Added the chnges :
```
Use virtctl to connect to the newly create VirtualMachineInstance.
```

iranzo commented:
** is used instead of ropper headings "##" please, replace what is a header with  proper heading level, and leave ** for things like notes, etc

iranzo commented on _posts/2019-05-15-How-To-Guide-Import-VM-Kubevirt.markdown:
````diff
@@ -0,0 +1,448 @@
+        name: cloudinitdisk
+
+```
+**Note**: 
````
There's a new note very close, should those be merged together?

DirectedSoul1 replied:
Yes, very good suggestion, I have merged both the notes into one.

iranzo commented on _posts/2019-05-15-How-To-Guide-Import-VM-Kubevirt.markdown:
```diff
@@ -0,0 +1,448 @@
+**Note**:
+
+- More examples of a VM declared as a `yaml` manifest can be seen [here](https://github.com/kubevirt/kubevirt/tree/master/cluster/examples)
+
```
Remove spaces between "-" items

DirectedSoul1 replied:
Removed the new Lines.
```
- More examples of a VM declared as a `yaml` manifest can be seen [here](https://github.com/kubevirt/kubevirt/tree/master/cluster/examples)
- Please feel free to take a look at how to deploy VM as a K8s object by using kubevirt add-on using minikube [here](https://kubevirt.io//quickstart_minikube/)
```

iranzo commented on _posts/2019-05-15-How-To-Guide-Import-VM-Kubevirt.markdown:
````diff
@@ -0,0 +1,448 @@
+```shell
+$ kubectl create -f vm.yaml
+virtualmachine "vm-cirros" created
+```
````
Leave extra new line between code blocks and next paragraph and same for others before it

DirectedSoul1 replied:
Added the new lines.

iranzo commented on _posts/2019-05-15-How-To-Guide-Import-VM-Kubevirt.markdown:
````diff
@@ -0,0 +1,448 @@
+$ kubectl get vmis
+No resources found.
+```
+Let’s start the VirtualMachine:
````
newline before

DirectedSoul1 replied:
Added the new line.
```
$ kubectl get vmis
No resources found.
```

Let’s start the VirtualMachine:

iranzo commented on _posts/2019-05-15-How-To-Guide-Import-VM-Kubevirt.markdown:
```diff
@@ -0,0 +1,448 @@
+The cdi-uploadproxy service must be accessible from outside the cluster. Here are some ways to do that:
+
+- [NodePort Service](https://kubernetes.io/docs/concepts/services-networking/service/#nodeport)
+
```
Remove extra newline

DirectedSoul1 replied:
Removed the new-lines.

iranzo commented on _posts/2019-05-15-How-To-Guide-Import-VM-Kubevirt.markdown:
```diff
@@ -0,0 +1,448 @@
+- [NodePort Service](https://kubernetes.io/docs/concepts/services-networking/service/#nodeport)
+
+- [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/)
+
```
Same here

DirectedSoul1 replied:
Fixed the new-line spaces.

iranzo commented on _posts/2019-05-15-How-To-Guide-Import-VM-Kubevirt.markdown:
```diff
@@ -0,0 +1,448 @@
+
+The supported image formats are:
+
+- .img
```
Use codeblocks for extension and add in 238 the comment about compressed

DirectedSoul1 replied:
Added code Blocks.
```
- .img

- .iso

- .qcow2

- Compressed .tar, .gz and .xz of the above formats are supported.
```

iranzo commented:
In general, I see that this PR is not based on latest suggestions/changes performed on the previous PR, please, do check them as starting point.

Thanks!

karmab commented:
can you address the issues @DirectedSoul1  ?

DirectedSoul1 commented:
> ** is used instead of ropper headings "##" please, replace what is a header with proper heading level, and leave ** for things like notes, etc

Added the ## and removed the "**" for the headings.

karmab commented on _posts/2019-05-15-How-To-Guide-Import-VM-Kubevirt.markdown:
```diff
@@ -0,0 +1,452 @@
+
+# Importing a VM into the Kubernetes Native Environment
+
+Motivation: Kubernetes have become the new way to orchestrate the containers and to handle the microservice architecture, but what if I already have legacy apps running on my old VM's in my datacenter ? Can those apps ever be made k8s friendly ? Well if that is the usecase for you then we have a solution now with kubevirt.
```
- kubernetes has become
- and handle the microservices
i wouldnt use the term legacy, just apps (or older apps)

DirectedSoul1 replied:
Made the changes as suggested. Thanks for pointing this out @karmab

karmab commented on _posts/2019-05-15-How-To-Guide-Import-VM-Kubevirt.markdown:
```diff
@@ -0,0 +1,452 @@
+
+Motivation: Kubernetes have become the new way to orchestrate the containers and to handle the microservice architecture, but what if I already have legacy apps running on my old VM's in my datacenter ? Can those apps ever be made k8s friendly ? Well if that is the usecase for you then we have a solution now with kubevirt.
+
+In this blog post we will discuss about how to deploy VM as a yaml template and steps on how to import it as a [PVC](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) onto your kubernetes environment using the CDI and kubevirt add-ons.
```
we will show how to deploy a vm

DirectedSoul1 replied:
Edited as suggested.

karmab commented on _posts/2019-05-15-How-To-Guide-Import-VM-Kubevirt.markdown:
```diff
@@ -0,0 +1,452 @@
+
+- A basic understanding of the k8s architecture: In its simplest terms Kubernetes is a portable, extensible open-source platform for managing containerized workloads and services, that facilitates both declarative configuration and automation. It has a large, rapidly growing ecosystem. Kubernetes services, support, and tools are widely available. For complete details check [Kubernetes-architectur ...
+
+- User is familiar with the concept of a [virsh based VM](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/virtualization_deployment_and_administration_guide/sect-guest_virtual_machine_installation_overview-creating_guests_with_virt_install)
```
of a libvirt based vm, rather ?

DirectedSoul1 replied:
Made the changes.

karmab commented on _posts/2019-05-15-How-To-Guide-Import-VM-Kubevirt.markdown:
```diff
@@ -0,0 +1,452 @@
+
+- User is familiar with the concept of a [virsh based VM](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/virtualization_deployment_and_administration_guide/sect-guest_virtual_machine_installation_overview-creating_guests_with_virt_install)
+
+- PersistentVolume (PV) is a piece of storage in the cluster that has been provisioned by an administrator. It is a resource in the cluster just like a node is a cluster resource. PVs are volume plugins like Volumes, but have a lifecycle independent of any individual pod that uses the PV. This API object captures the details of the implementation of the storage, be that NFS, iSCSI, or a cloud-pro ...
```
i dont think we need to explain what a pv/pvc is, worst case a single sentence and a link to the documentation

DirectedSoul1 replied:
Taken care of( by putting a single sentence and an extended link.

karmab commented on _posts/2019-05-15-How-To-Guide-Import-VM-Kubevirt.markdown:
```diff
@@ -0,0 +1,452 @@
+
+# Virtual machines:
+
+**Virtual machines:**
```
i believe we dont need to explain (again) what a vm does and how to handle it, user guide contains enough information

DirectedSoul1 replied:
Are you suggesting to remove all of these :
```
# Virtual machines:

**Virtual machines:**
A VirtualMachine provides additional management capabilities to a VirtualMachineInstance inside the cluster. That includes:

- ABI stability

- Start/stop/restart capabilities on the controller level

- Offline configuration change with propagation on VirtualMachineInstance recreation

- Ensure that the VirtualMachineInstance is running if it should be running

It focuses on a 1:1 relationship between the controller instance and a virtual machine instance. In many ways it is very similar to a StatefulSet with spec.replica set to 1.
```

DirectedSoul1 replied:
Added a one-liner and pointed an external link for more.

DirectedSoul1 replied:
Deleted non-relevant concepts.

karmab commented on _posts/2019-05-15-How-To-Guide-Import-VM-Kubevirt.markdown:
```diff
@@ -0,0 +1,452 @@
+
+A VirtualMachine will make sure that a VirtualMachineInstance object with an identical name will be present in the cluster, if `spec.running` is set to `true`. Further it will make sure that a VirtualMachineInstance will be removed from the cluster if `spec.running` is set to `false`.
+
+There exists a field `spec.runStrategy` which can also be used to control the state of the associated VirtualMachineInstance object. To avoid confusing and contradictory states, these fields are mutually exclusive. An extended explanation of `spec.runStrategy` vs `spec.running` can be found in [Run Strategies](https://kubernetes.io/docs/tasks/run-application/run-single-instance-stateful-applicati ...
```
is this field especially relevant in the context of importing a vm. if yes, let s indicate why

DirectedSoul1 replied:
It's not really important in the context of VM import, but it does explain as to how the VMI itself gets started in the kubevirt environment. Should this be removed?

DirectedSoul1 replied:
removed the `spec.runStrategy` field, as it is not really needed in this context.

karmab commented on _posts/2019-05-15-How-To-Guide-Import-VM-Kubevirt.markdown:
```diff
@@ -0,0 +1,452 @@
+
+This document for now deals with the third use case and covers the HTTP based import use case at the end of the BlogPost. So you should have CDI installed in your cluster, a VM disk that you’d like to upload, and virtctl in your path.
+
+Lets begin by installing the latest CDI release [here](https://github.com/kubevirt/containerized-data-importer/releases) (currently v1.9.0)
```
i think we dont need to indicate how to install cdi, it s documented on cdi web page

DirectedSoul1 replied:
Makes sense, removed the paragraph about CDI explanation. But I have pointed the reader to the GitHub link of CDI.

karmab commented on _posts/2019-05-15-How-To-Guide-Import-VM-Kubevirt.markdown:
```diff
@@ -0,0 +1,452 @@
+
+The above method explains the import of the VM by importing the locally available image, now lets see the concept of DataVolumes:
+
+# DataVolume:
```
does using datavolumes improve the importing process. if yes, let s outline why. If not, i dont know if we want to talk about data volumes

DirectedSoul1 replied:
It does not improve the process, but it gives a good idea of the whole `yaml` structure, i do feel to get a complete picture this section is needed.

iranzo commented:
Rebase this one too to fix the issue with the link that is fixed in master please

ptrnull commented:
/lgtm
/approve
